source/content.js

```diff
@@ -54,6 +55,30 @@ function onSingleTweetOpen(cb) {
 	}, {attributes: true});
 }
 
+function onDMDelete() {
+	observeEl('body', async mutations => {
+		const savedMessages = await browser.storage.local.get();
+		const pendingRemoval = [];
+
+		for (const mutation of mutations) {
+			if (mutation.target.id === 'confirm_dialog') {
```

Comment:
Instead of a `for` and an `if`, `.filter` the `mutations` array and apply the callback to all items that pass the filter, if any.

Reply:
You are correct. I missed the `break` on line 74! Since you have the `break` statement there, the `for..of` is, indeed, the best choice! Sorry!

Reply:
No worries! 😊